storage/replica_raftstorage.go

```diff
@@ -112,16 +114,32 @@ func (r *Replica) Entries(lo, hi, maxBytes uint64) ([]raftpb.Entry, error) {
 	// If neither the number of entries nor the size limitations had an
 	// effect, we weren't able to supply everything the client wanted.
 	if len(ents) != int(hi-lo) && (maxBytes == 0 || size < maxBytes) {
```

Comment:
The check for `ErrCompacted` should come before the one for `maxBytes`. The original check was kind of sloppy and could cover up missing records if the `maxBytes` limit was hit. The sequence should be
* If `lo` has been compacted, return `ErrCompacted`
* If `ents` contains gaps, return an error (doesn't really matter what)
* If the last entry is not `hi-1` and `size < maxBytes`, return `ErrUnavailable`